backend/notifications/notificationJob.js

```diff
@@ -0,0 +1,57 @@
+import { schedule } from "node-cron";
+import { prisma } from "../utils/helpers.js";
+import { generateAllNotifications } from "./notificationGenerator.js";
+import { evaluateDecisionTree } from "./decisionTreeEvaluator.js";
+
+const processUserNotifications = async (userId, io) => {
+  const userInterest = await prisma.userInterest.findUnique({
+    where: { userId },
+    include: { interests: true },
+  });
+
+  if (!userInterest) return;
+
+  const generatedNotifications = await generateAllNotifications(userId);
+
+  for (const notification of generatedNotifications) {
+    await prisma.notification.create({
```

Comment:
i'm wondering if we still need to store all the notifications, even the ones that are unsent - if we're only sending one a day, should we only store that one? or will there be a separate section where users can see all their notifications, even the ones that aren't sent through cron job?

Reply:
I think you're right about not having to store every single notification. I will work on this when I am working on stretch features.

Reply:
in my opinion, we should store all notifs so user can see past notifs (similar to facebook)